indexing-solr/src/main/java/edu/unc/lib/boxc/indexing/solr/filter/SetDatastreamFilter.java

```diff
@@ -128,29 +128,125 @@ private String getExtent(List<BinaryObject> binList) {
         }
     }
 
+    private String formatVideoExtent(List<Element> trackInfo, String pid) {
+        var numTracks = trackInfo.size();
+        var videoTrack = 0;
+        var audioTrack = 0;
+
+        if (numTracks > 1) {
+            for (int i = 0; i < numTracks; i++) {
+                var type = trackInfo.get(i).getAttributeValue("type");
+                if (type.equals("video")) {
+                    videoTrack = i;
+                }
+                if (type.equals("audio")) {
+                    audioTrack = i;
+                }
+            }
+        }
+
+        var videoInfo = trackInfo.get(videoTrack);
+        var videoTime = formatTime(videoInfo);
+        var videoHeight = videoInfo.getChildTextTrim("width", FITS_NS);
+        var videoWidth = videoInfo.getChildTextTrim("height", FITS_NS);
+
+        // Some videos have separate tracks for the audio content
+        String audioTime = null;
+        if (numTracks > 1) {
+            audioTime = formatTime(trackInfo.get(audioTrack));
+        }
+
+        // Audio and video don't necessarily have the same play times, so select the larger one if both are present
+        String trackTime = (audioTime == null || Integer.parseInt(videoTime) >= Integer.parseInt(audioTime))
+                ? videoTime : audioTime;
+        var extent = formatExtent(videoHeight, videoWidth, pid);
+        return (extent == null) ? "xx" + trackTime : extent + "x" + trackTime;
+    }
+
+    private String formatTime(Element durationElement) {
+        var durationMilliseconds = durationElement.getChild("milliseconds", FITS_NS);
+        var duration  = durationElement.getChild("duration", FITS_NS);
+
+        if (durationMilliseconds != null) {
+            return normalizeTime(durationMilliseconds.getTextTrim());
+        }
+
+        if (duration != null) {
+            return normalizeTime(duration.getTextTrim());
+        }
+
+        return "-1";
+    }
+
+    private String normalizeTime(String duration) {
+        String output = "";
+        Pattern pattern = Pattern.compile("\\d+:\\d+:\\d+", Pattern.CASE_INSENSITIVE);
+        Matcher matcher = pattern.matcher(duration);
+        boolean matchFound = matcher.find();
+
+        if (matchFound) {
+            var durationParts = duration.split(":");
+            var hoursToSeconds = Integer.parseInt(durationParts[0]) * 60 * 60;
+            var minutesToSeconds = Integer.parseInt(durationParts[1]) * 60 ;
+            var seconds = Integer.parseInt(durationParts[2]);
+
+            output += hoursToSeconds + minutesToSeconds + seconds;
+        } else {
+            output += millisecondsToSeconds(duration);
+        }
+
+        return output;
+    }
+
+    private Integer millisecondsToSeconds(String duration) {
+        var durationToSeconds = Integer.parseInt(duration) / 1000.0;
+        return (int) Math.ceil(durationToSeconds);
+    }
+
+    private String formatExtent(String imgHeight, String imgWidth, String id) {
```

Comment:
Its `id` in this method and `pid` in some of the others. Might as well make it consistent with one or the other. `pid` might be clearer about what kind of id it is?